frontend/packages/topology/src/components/graph-view/Topology.tsx

```diff
@@ -138,10 +143,14 @@ const Topology: React.FC<TopologyProps &
   const componentFactoryExtensions = useExtensions<TopologyComponentFactory>(
     isTopologyComponentFactory,
   );
+  const [dynamicComponentFactoryExtensions, isFactoryComponentsResolved] = useResolvedExtensions<
+    DynamicTopologyComponentFactory
+  >(isDynamicTopologyComponentFactory);
```

Comment:
Can we move away from `useExtensions` for the old extension and make use of `useResolvedExtensions`?
Then combine the results two variables: one for the resolved state and one for an array of factories.

This way it'll be two code paths for setup but consumption will be simple.